packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -111,21 +146,55 @@ class FlutterPlugin implements Plugin<Project> {
         return result
     }
 
+    /**
+     * Returns the platform that is used to extract the `libflutter.so` and the .class files.
+     *
+     * Note: This is only needed to add the .class files.
+     * Unfortunately, the engine artifacts include the .class and libflutter.so files.
+     */
+    private String getBasePlatform(Project project) {
+        if (PLATFORM_ARM64 in getTargetPlatforms(project)) {
+            return PLATFORM_ARM64;
+        }
+        return PLATFORM_ARM32;
+    }
+
     @Override
     void apply(Project project) {
         project.extensions.create("flutter", FlutterExtension)
         project.afterEvaluate this.&addFlutterTask
 
-        allTargetPlatforms.each { currentTargetPlatformValue, abi ->
+        // By default, assembling APKs generates fat APKs if multiple platforms are passed.
+        // Configuring split per ABI allows to generate separate APKs for each abi.
+        // This is a noop when building a bundle.
+        if (this.splitPerAbi(project)) {
             project.android {
-                packagingOptions {
-                    pickFirst "lib/${abi}/libflutter.so"
-
-                    // Disable warning by *-android-strip: File format not recognized
-                    doNotStrip "*/${abi}/lib_vm_snapshot_data.so"
-                    doNotStrip "*/${abi}/lib_vm_snapshot_instr.so"
-                    doNotStrip "*/${abi}/lib_isolate_snapshot_data.so"
-                    doNotStrip "*/${abi}/lib_isolate_snapshot_instr.so"
+                splits {
+                    abi {
+                        // Enables building multiple APKs per ABI.
+                        enable true
+                        // By default all ABIs are included, so use reset() and include to specify that we only
```

Comment:
I think this is two comments as one (we probably only need to keep the "Resets the list of ABIs ..." line.